llvm/lib/FileCheck/FileCheck.cpp

```diff
@@ -2468,13 +2468,16 @@ FileCheckString::CheckDag(const SourceMgr &SM, StringRef Buffer,
 
 static bool ValidatePrefixes(StringRef Kind, StringSet<> &UniquePrefixes,
                              ArrayRef<StringRef> SuppliedPrefixes) {
+  static const char *Directives[] = {"-NEXT",  "-SAME", "-EMPTY", "-NOT",
+                                     "-COUNT", "-DAG",  "-LABEL"};
+
   for (StringRef Prefix : SuppliedPrefixes) {
     if (Prefix.empty()) {
       errs() << "error: supplied " << Kind << " prefix must not be the empty "
              << "string\n";
       return false;
     }
-    static const Regex Validator("^[a-zA-Z0-9_-]*$");
+    static const Regex Validator("^[a-zA-Z][a-zA-Z0-9_-]*$");
```

Comment:
This change was also picking up prefixes that were just numbers (e.g. 32 and 64 for 32 vs 64 bit tests)- should we allow those?

Reply:
Well, yes, but error message says "prefix must start with a letter" so this existed before that PR?

Reply:
Good point. Would you mind updating the documentation at least? Perhaps make a general section about identifiers and in the prefix and variable sections mention that those must be valid identifiers.

Reply:
Being able to have a prefix like "32" or "64" is quite useful for LLVM binutils tests where the thing that is different is whether the file format is 32-bit or 64-bit. There's no reason to forbid those patterns. If necessary, update the message and/or docs to match the current behaviour rather than changing the behaviour.

Reply:
Currently i don't see any documented (https://llvm.org/docs/CommandGuide/FileCheck.html) restrictions for prefixes except that error message, which was there more than last 8(?) years, regex was added somewhere later in https://github.com/llvm/llvm-project/commit/2aa0217add1adbcabbc16e3ec44937e4277e117b but probably relaxed to not to fix possible breakage at that time, idk (or maybe just bug).
> Being able to have a prefix like "32" or "64" is quite useful

Noone forbids to write something like x32bit (except this need bless bunch of tests, i see this as main reason against that change).

Reply:
I hope that check prefixes like `# 64: ` can be kept working.

There are quite a few occurrences: `rg '^(;|#|//) \d+:' llvm/test/ -l`

Reply:
OKay.